apps/wallet/src/ui/app/pages/home/transfer-coin/index.tsx

```diff
@@ -172,67 +167,98 @@ function TransferCoinPage() {
+            { setSubmitting }: FormikHelpers<FormValues>
+        ) => {
+            setCurrentStep((prev) => prev + 1);
+            setFormData(formData);
+            setSubmitting(false);
+        },
+        []
+    );
+
     if (!coinType) {
         return <Navigate to="/" replace={true} />;
     }
 
     const StepOneForm = (
-        <Formik
-            initialValues={formData}
-            validateOnMount={true}
-            validationSchema={validationSchemaStepOne}
-            onSubmit={handleNextStep}
-        >
-            <StepOne
-                coinSymbol={coinSymbol}
-                coinType={coinType}
-                onClearSubmitError={handleOnClearSubmitError}
-                onAmountChanged={(anAmount) => setAmountToSend(anAmount)}
-            />
-        </Formik>
-    );
+        <div className="flex flex-col w-full mt-2.5">
+            <div className="mb-7">
+                <ActiveCoinsCard activeCoinType={coinType} />
+            </div>
 
-    const StepTwoForm = (
-        <Formik
-            initialValues={formData}
-            validateOnMount={true}
-            validationSchema={validationSchemaStepTwo}
-            onSubmit={onHandleSubmit}
-        >
-            <StepTwo
-                submitError={sendError}
-                coinSymbol={coinSymbol}
-                coinType={coinType}
-                gasBudgetEstimation={gasBudgetEstimation || null}
-                gasCostEstimation={gasBudgetEstimation || null}
-                gasEstimationLoading={isLoading}
-                onClearSubmitError={handleOnClearSubmitError}
-            />
-        </Formik>
+            <Formik
+                initialValues={formData}
+                validationSchema={validationSchemaStepOne}
+                onSubmit={handleNextStep}
+                enableReinitialize={true}
+            >
+                <StepOne
+                    submitError={sendError}
+                    coinType={coinType}
+                    balance={coinBalance}
+                    gasCostEstimation={gasBudgetEstimation || null}
+                    onClearSubmitError={handleOnClearSubmitError}
+                    onAmountChanged={(anAmount) => setAmountToSend(anAmount)}
+                />
+            </Formik>
+        </div>
     );
 
-    const steps = [StepOneForm, StepTwoForm];
-
-    const SendCoin = (
-        <div className={st.container}>
-            <PageTitle
-                title="Send Coins"
-                back={currentStep > 1 ? handleBackStep : '/'}
-            />
-
-            <Content className={st.content}>
-                <Loading loading={loadingBalance}>
-                    <ProgressBar
-                        currentStep={currentStep}
-                        stepsName={['Amount', 'Address']}
+    const StepTwoPreview = (
+        <BottomMenuLayout>
+            <Content>
+                <div className="flex flex-col w-full mt-2.5">
+                    <StepTwo
```

Comment:
Same for this one rename it to something like `SendTokensReview` and move all related functionality inside the component. Probably also this and StepOne can share the same container/layout (`<div className="flex flex-col w-full mt-2.5">` seems to be the same)